qadence/transpile/noise.py

```diff
@@ -23,24 +24,26 @@ def _set_noise(
 
 
 def set_noise(
-    circuit: QuantumCircuit | AbstractBlock,
+    circuit: QuantumCircuit | AbstractBlock | ConvertedCircuit,
     noise: NoiseHandler | None,
     target_class: AbstractBlock | None = None,
 ) -> QuantumCircuit | AbstractBlock:
     """
     Parses a `QuantumCircuit` or `CompositeBlock` to add noise to specific gates.
 
+    If `circuit` is a `ConvertedCircuit`, this is done within `circuit.abstract`.
+
     Changes the input in place.
 
     Arguments:
         circuit: the circuit or block to parse.
         noise: the NoiseHandler protocol to change to, or `None` to remove the noise.
         target_class: optional class to selectively add noise to.
     """
-    is_circuit_input = isinstance(circuit, QuantumCircuit)
-
-    input_block: AbstractBlock = circuit.block if is_circuit_input else circuit  # type: ignore
+    to_convert = circuit.abstract if isinstance(circuit, ConvertedCircuit) else circuit
+    is_circuit_input = isinstance(to_convert, QuantumCircuit)
 
-    output_block = apply_fn_to_blocks(input_block, _set_noise, noise, target_class)
+    input_block: AbstractBlock = to_convert.block if is_circuit_input else to_convert  # type: ignore
```

Comment:
Uff...I feels quite a complicated bit of logic that might be simplified ? I don't have a straight answer off the top of my head but it really feels that you only need the `.block` when this is a `QuantumCircuit` and the `circuit` otherwise.